### PR TITLE
adding FizzBuzz and Series.py

### PR DESCRIPTION
**Pull request by pythonwa**
Hi Professor,
I changed FizzBuzz with 3.6 version and follow Python convention. I also spent a quite amount of time for Series assignment, I hope you are happy with it. :-)

### CODE REVIEW

**Comment by pythonwa** [comment]
Hi Professor, I just rewrote the second assignment by using slicing. Thanks.

**Comment by pythonwa** [comment]
Hi Professor, I just submitted strformat_lab.py.

**Comment by natasha-aleksandrova** [comment]
sounds good!

File: students/tonylee/SecondAssignment.py [hunk 1]
```diff
@@ -1,69 +1,35 @@
...
 
-print(remove_first4_last4_every_other_item(a_tuple))
+print(remove_first4_last4_every_other_item(a_string))
+print(reverse_element(a_string))
 
-print mid_last_first(a_string)
-print mid_last_first(a_tuple)
\ No newline at end of file
+print(mid_last_first(a_string))
+print(mid_last_first(a_tuple))
```
Comment (natasha-aleksandrova) [inline students/tonylee/SecondAssignment.py]: 👍 much better on slicing!

**Comment by pythonwa** [comment]
Hi Professor, I just submitted mailroom2. Thank.

File: students/tonylee/Series.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
+fruitList = ["Apples", "Pears", "Oranges", "Peaches"]
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: don't forget per pep8 guideline - snake_case -> `fruit_list`

File: students/tonylee/Series.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+def print_fruit_list(fruitList):
+    print(fruitList)
+
+def get_fruit_name(index):
+    print("Fruit[" + str(index) + "] = " + fruitList[index-1])
+
+def get_fruit_number():
+	  prompt = "Enter the fruit's number in the fruit list: "
+	  fruitIndex = int(input(prompt))
+	  return fruitIndex
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: overindented - should be 4 spaces

File: students/tonylee/Series.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+def add_fruit_at_front_with_insert():
+    newFruit = get_new_fruit()
+    fruitList.insert(0, newFruit)
+    print_fruit_list(fruitList)
+
+def display_all_fruits_begin_with_p():
+    print("List of all fruits that begin with \"P\": ", end= " ")
+    for fruit in fruitList:
+        if fruit[:1] == 'P':
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: small bug: should be `if fruit[0] == 'P'`

Comment (natasha-aleksandrova) [reply students/tonylee/Series.py]:
```
>>> f = ["Apples", "Pears", "Oranges", "Peaches"]
>>> f[:1]
['Apples']
```

File: students/tonylee/Series.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      prompt = "Do you like " + fruitName.lower() + " ?"
+      answer = input(prompt)
+      return answer
+
+def isValidAnswer(answer):
+      validAnswer = ["yes", "no"]
+      if answer in validAnswer:
+          return True
+      else:
+          return False
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: above two functions over-indented.

File: students/tonylee/Series.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+def get_answer(fruitName):
+      prompt = "Do you like " + fruitName.lower() + " ?"
+      answer = input(prompt)
+      return answer
+
+def isValidAnswer(answer):
+      validAnswer = ["yes", "no"]
+      if answer in validAnswer:
+          return True
+      else:
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: shortcut:
```
def is_valid_answer(answer):
    valid_answer = ("yes", "no")  # use tuples for immutable data
    return answer in valid_answer
```

File: students/tonylee/Series.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+def series4():
+    print_fruit_list(copy_and_reverse_fruit_list())
+    remove_last_fruit()
+    print_fruit_list(fruitList)
+
+series1()
+series2()
+series3()
+series4()
```
Comment (natasha-aleksandrova) [inline students/tonylee/Series.py]: some small pep8 violations, otherwise great work!

File: students/tonylee/strformat_lab.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    for i in test_tuple:
+      print("{:5}".format(i))
+
+task_one()
+task_two()
+task_three()
+task_four()
+task_five()
+task_six()
+bonus_task()
```
Comment (natasha-aleksandrova) [inline students/tonylee/strformat_lab.py]: 👍

File: students/tonylee/mailroom.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    numGifts[donor] = numGifts[donor] + 1
+
+while True:
+    choice = input("Please choose from a menu of 3 actions: 'Send a Thank You', 'Create a Report' or 'quit': ")
+    if choice == "Send a Thank You":
+        send_thankYou()
+    elif choice == "Create a Report":
+        create_report()
+    elif choice == "quit":
+        break
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom.py]: couple notes:
* put the while True block into `if __name__ == "__main__"` block
* it is best to use numbers for user options - having user type strings is not user friendly and error prone.

File: students/tonylee/mailroom.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        totalDonationAmount.append(0.0)
+        numGifts.append(0)
+        setDonation(len(donors)-1, getDonation())
+
+    print(f'Thank you {donorName}, for your donation!')
+
+def create_report():
+    print('{:20}|{:15}|{:12}|{}'.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    for i in range(len(donors)):
+        print('{:20} ${:10,.2f}{:^17} ${:10,.2f}'.format(donors[i], totalDonationAmount[i], numGifts[i], totalDonationAmount[i]/numGifts[i]))
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom.py]: this function is missing sorting feature.

File: students/tonylee/mailroom.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+def isSameDonor(donorName):
+    for i in range(len(donors)):
+        if donors[i] == donorName:
+            setDonation(i, getDonation())
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom.py]: this should be done outside of this function - it is best practice for functions to have side-effects, meaning:

if function implies that it returns a boolean then it should be a function that only does; here it is doing a boolean check + modifies that data.

File: students/tonylee/mailroom.py [hunk 12]
```diff
@@ -0,0 +1,48 @@
+donors = ["Tony Lee", "Andy Arko", "Michelle Lee", "Tom Ludwinski", "Kurt Lafser"]
+totalDonationAmount = [100, 200, 300, 400, 500]
+numGifts = [1, 1, 1, 1, 1]
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom.py]: lots of pep8 naming violations.

**Comment by natasha-aleksandrova** [comment]
@pythonwa I'm not going to approve the mailroom2 (lesson 4?) at the moment.

Please review the instructions again, lesson 4 is about dictionaries and file writing (which I see).

* you should use dict for the main data structure
* you should use dict as a switch statement for main entry point
* you should try using dict for string formatting
* add missing report sorting feature
* add __name__ == "__main__" guard
* apply fixes from mailroom lesson3 review

let me know if you have any questions on anything above, thanks!

**Pull request merged**